### ISSUE TITLE
I want to discuss with developers why qsqltablemodel is used instead of qsqlquery

### ISSUE THREAD

**Issue by moshengrenzenme** [ukui/ukui-sidebar#53]
我查到的资料QSqlTableModel一般和表格联动，反馈这里好像没有表格，使用QSqlQuery虽然代码多一点但是可以更容易读懂吧，毕竟都是sql，用QSqlTableModel写的有点难懂还不是很容易控制，如果用原生的sql可以控制的更精确，请开发有时间回复一下

**Comment by whailiang** [comment]
是的，我也是第一次使用QsqlTableModel 也发现了这个问题，现在缕清逻辑后，发现在插件中使用数据库的话，如果其他插件也使用数据库在使用qtcreator编译时，会报可能会出现内存泄漏的警告，所以调整了代码，反馈这里 没有必要使用数据库，可以直接使用qstringlist 来保存数据。

**Comment by whailiang** [comment]
> 我查到的资料QSqlTableModel一般和表格联动，反馈这里好像没有表格，使用QSqlQuery虽然代码多一点但是可以更容易读懂吧，毕竟都是sql，用QSqlTableModel写的有点难懂还不是很容易控制，如果用原生的sql可以控制的更精确，请开发有时间回复一下

**Comment by whailiang** [comment]
> 我查到的资料QSqlTableModel一般和表格联动，反馈这里好像没有表格，使用QSqlQuery虽然代码多一点但是可以更容易读懂吧，毕竟都是sql，用QSqlTableModel写的有点难懂还不是很容易控制，如果用原生的sql可以控制的更精确，请开发有时间回复一下

是的 你说的很对 qsqltablemodel 是比较难懂，而且现在feedback放入侧边栏中还有会数据库的问题，因为可能和其他插件的数据库有冲突吧，所以现在重构之后 已经不再用数据库了，好像没什么必要用数据库，用list就可以正常使用了。qsqlquery就像你说的直接调用sql语句 当然更容易读懂，但是因为我sql语句不太熟就没用这个

**Comment by moshengrenzenme** [comment]
@whailiang sql这东西我会，如果有时间了，可以探讨一下，正好我qt不行（我差点就把你的数据库操作给改了）

**Comment by whailiang** [comment]
@moshengrenzenme 可以探讨一下
